File: expandablrecyclerview/src/main/java/com/umut/expandablrecyclerview/adapter/holder/ParentViewHolder.java
package com.umut.expandablrecyclerview.adapter.holder;


import android.view.View;

import androidx.annotation.NonNull;

import java.util.List;

import static com.umut.expandablrecyclerview.adapter.ExpandableViewAdapter.State;

public abstract class ParentViewHolder extends ExpandableViewHolder<Integer> {

    public ParentViewHolder(View itemView) {
        super(itemView);
    }

    /**
     * binds the value to view holder
     *
     * @param coordinate Coordinate of item
     * @param state      state of parent (expanded or collapsed)
     */
    public abstract void bind(int coordinate, @State int state);

    public boolean onStateChanged(int coordinate, @State int state) {
        return false;
    }

    public boolean update(@NonNull List<Object> payload, int coordinate, @State int state) {
        return false;
    }
}

File: expandablrecyclerview/src/main/java/com/umut/expandablrecyclerview/adapter/holder/ParentViewHolder.kt
package com.umut.expandablrecyclerview.adapter.holder


import android.view.View

import com.umut.expandablrecyclerview.adapter.ExpandableViewAdapter.State

abstract class ParentViewHolder(itemView: View) : ExpandableViewHolder<Int>(itemView) {

    /**
     * binds the value to view holder
     *
     * @param coordinate Coordinate of item
     * @param state      state of parent (expanded or collapsed)
     */
    abstract fun bind(coordinate: Int, @State state: Int)

    fun onStateChanged(coordinate: Int, @State state: Int): Boolean {
        return false
    }

    fun update(payload: List<Any>, coordinate: Int, @State state: Int): Boolean {
        return false
    }
}
